pkg/languagedetection/detector.go

```diff
@@ -0,0 +1,90 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// LanguageName is a string enum that represents a detected language name.
+type LanguageName string
+
+const (
+	Node    LanguageName = "node"
+	Dotnet  LanguageName = "dotnet"
+	Python  LanguageName = "python"
+	Java    LanguageName = "java"
+	Unknown LanguageName = ""
+)
+
+// Language contains metadata collected from the call to `DetectLanguage`
+type Language struct {
+	Name LanguageName
+}
+
+type languageFromCLI struct {
+	name      LanguageName
+	validator func(exe string) bool
+}
+
+// knownPrefixes maps languages names to their prefix
+var knownPrefixes = map[string]languageFromCLI{
+	"python": {name: Python},
+	"java": {name: Java, validator: func(exe string) bool {
+		if exe == "javac" {
+			return false
+		}
+		return true
+	}},
```

Comment:
shouldn't add node/dotnet here ?

Reply:
I wanted to be careful with that. I don't know of any cases where there is a version suffix on the end of a node or dotnet binary. I did however find cases where there were processes like `node_validator`, which would not match correctly if we were to list `node*` as a prefix.

Please correct me if you know of any cases otherwise! 😄